client/trino-client/src/test/java/io/trino/client/auth/external/TestExternalAuthenticator.java

```diff
@@ -157,4 +249,44 @@ private static Response getUnauthorizedResponse(String challengeHeader, Request
                 .header(WWW_AUTHENTICATE, challengeHeader)
                 .build();
     }
+
+    static class ConcurrentRequestAssertion
+    {
+        private final List<Throwable> exceptions = new ArrayList<>();
+        private final List<Request> requests = new ArrayList<>();
+
+        public ConcurrentRequestAssertion(List<Future<Request>> requests)
```

Comment:
Can we use something like `com.google.common.util.concurrent.Futures#allAsList`? We should care about all results and all exception for all futures.